### DISCUSSION TITLE
Signed commit with a GPG RSA key still unverified for GitHub

### DISCUSSION THREAD

**Discussion by Zilog80a** [community/community#47102] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hello,

It's my first time posting here, so this post can be out of the product feedback scope or may not fit as a discussion, sorry by advance.

I wanted to try the gpg signed commit support in GitHub, so i add my usual RSA gpg public key on my GitHub account.
It's 2048 bits length key with SCEA roles :

```
sec  rsa2048/05502908DF7BF658
created: 2003-05-29  expires: never       usage: SCEA
```

I made a signed commit with this key on a local repos and check that git validate the signed commit :

```
$ git log --show-signature
commit 22796fe739b521eea49219eb910470fdcdef1a73 (HEAD -> ...)
gpg: Signature made sam. 11 févr. 2023 20:14:40
gpg:                using RSA key 94D3AA52F14EABF7330F885905502908DF7BF658
gpg: Good signature from ...
```

I then pushed this commit on my corresponding GitHub repository. I have set the "Flag unsigned commits as unverified" on my GitHub account. As one of the key uid email matches my primary account email and the other key uid matches the backup email, i was expecting this commit to appear with the "Verified" tag on the commit history of this repos, however it appears with the "Unverified" tag and the associated comment :

```
The key that signed this doesn't have usage flags that allow signing.
GPG key ID: 05502908DF7BF658
```
I've exported my key with `$ gpg -armor --export DF7BF658`, i've first looked if the _export_ option requires specifics flags to embed the key usages in the gpg documentation, without success.

I then look for similar issues in the community discussions and found few ones matching. In theses issues the poster was invited to contact GitHub support.

Before attempting that, i was wondering if i could miss something obvious, as i'm relatively new to GitHub. Maybe my key has to be signed by a third party or something like that ?

**Comment by airtower-luna** [comment]
For your commits to be recognized as verified you need to [add the public key to your account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account). I'm not sure if it'll apply retroactively (to the commit you already pushed), but definitely to any new ones.

**Reply by Zilog80a** [reply]
Thanks for your suggestion, however i've already added the related key to my GitHub account :

> so i add my usual RSA gpg public key on my GitHub account

I guess there is something with the key itself.

**Comment by DNin01** [comment]
[Generating a new GPG key on GitHub Docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) says that your key must be at least 4096 bits long, and it looks like your key was only 2048 bits long.
> 5. At the prompt, specify the key size you want, or press `Enter` to accept the default. Your key must be at least 4096 bits.

My guess is that your key was too short for GitHub, so try making a new one, ensuring you generate at least 4096 bits. Hopefully that solves your problem.

**Reply by Zilog80a** [reply]
That's probably that. I'll try that and confirm back. Thank you.
